ksqldb-rest-app/src/main/java/io/confluent/ksql/api/impl/InsertsStreamEndpoint.java

```diff
@@ -66,17 +68,20 @@ private DataSource getDataSource(
   ) {
     final DataSource dataSource = metaStore.getSource(sourceName);
     if (dataSource == null) {
-      throw new KsqlException("Cannot insert values into an unknown stream: "
-          + sourceName);
+      throw new KsqlApiException(
+          "Cannot insert values into an unknown stream: " + sourceName, ERROR_CODE_BAD_STATEMENT);
```

Comment:
Should this be ERROR_CODE_NOT_FOUND instead?

Reply:
I think BAD_STATEMENT is ok. NOT _FOUND is more if the uri doesn't exist.